monkey/monkey_island/cc/agent_event_handlers/scan_event_handler.py

```diff
@@ -56,11 +57,22 @@ def handle_tcp_scan_event(self, event: TCPScanEvent):
 
         try:
             target_machine = self._get_target_machine(event)
+            source_node = self._get_source_node(event)
 
             self._update_nodes(target_machine, event)
+            self._update_tcp_connections(source_node, target_machine, event)
         except (RetrievalError, StorageError, UnknownRecordError):
             logger.exception("Unable to process tcp scan data")
 
+    def _get_source_node(self, event: AbstractAgentEvent) -> Node:
+        machine = self._get_source_machine(event)
```

Comment:
```suggestion
    def _get_source_node(self, event: ScanEvent) -> Node:
        machine = self._get_source_machine(event)
```

Reply:
This method would work with any `AbstractAgentEvent`, because they all have `source`. I'm not sure where we should put all of utility methods like getting source node by event, getting machine by agent id, etc.